### PR TITLE
Replace Service Description Profiles Enum with an open List

### PR DESCRIPTION
**Pull request by sebbader-sap**

### CODE REVIEW

**Comment by sebbader-sap** [comment]
@danielporta I just tested it by manually translating it to JSON Schema. Summary: "oneOf" is wrong but "anyOf" does the job.

**Comment by danielporta** [comment]
@sebbader-sap: I don't think this will work. At least, the [openapi-spec-validator](https://github.com/python-openapi/openapi-spec-validator) did not succeed with my MWE. 

``` yaml

openapi: 3.0.3
info:
  title: MWE for anyOf validation
  description: a description
  contact:
    name: Industrial Digital Twin Association (IDTA)
    email: info@idtwin.org
  license: 
    name: CC BY 4.0
    url: https://creativecommons.org/licenses/by/4.0/
  version: V3.0.1_SSP-001
paths:
  /description:
    get:
      tags:
        - Description API
      summary: Returns the self-describing information of a network resource (ServiceDescription)
      operationId: GetDescription
      x-semanticIds:
        - https://admin-shell.io/aas/API/Descriptor/GetDescription/3/0
      responses:
        '200':
          description: Requested Description
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/ServiceDescription'
components: 
  schemas:   
    ServiceDescription: 
      description: The Description object enables servers to present their capabilities to the clients, in particular which profiles they implement. At least one defined profile is required. Additional, proprietary attributes might be included. Nevertheless, the server must not expect that a regular client understands them.
      example: >-
        {
          "profiles": [
            "https://admin-shell.io/aas/API/3/0/AssetAdministrationShellRegistryServiceSpecification/SSP-002",
            "https://admin-shell.io/aas/API/3/0/SubmodelRegistryServiceSpecification/SSP-002"
          ]
        }
      properties: 
        profiles:
          type: array
          minItems: 1
          items:
            type: string
            anyOf:
              - const: "https://admin-shell.io/aas/API/3/0/AssetAdministrationShellServiceSpecification/SSP-001"
              - const: "https://admin-shell.io/aas/API/3/0/AssetAdministrationShellServiceSpecification/SSP-002"
              - const: "https://admin-shell.io/aas/API/3/0/SubmodelServiceSpecification/SSP-001"
              - const: "https://admin-shell.io/aas/API/3/0/SubmodelServiceSpecification/SSP-002"
              - const: "https://admin-shell.io/aas/API/3/0/SubmodelServiceSpecification/SSP-003"
              - const: "https://admin-shell.io/aas/API/3/0/AasxFileServerServiceSpecification/SSP-001"
              - const: "https://admin-shell.io/aas/API/3/0/AssetAdministrationShellRegistryServiceSpecification/SSP-001"
              - const: "https://admin-shell.io/aas/API/3/0/AssetAdministrationShellRegistryServiceSpecification/SSP-002"
              - const: "https://admin-shell.io/aas/API/3/0/SubmodelRegistryServiceSpecification/SSP-001"
              - const: "https://admin-shell.io/aas/API/3/0/SubmodelRegistryServiceSpecification/SSP-002"
              - const: "https://admin-shell.io/aas/API/3/0/DiscoveryServiceSpecification/SSP-001"
              - const: "https://admin-shell.io/aas/API/3/0/AssetAdministrationShellRepositoryServiceSpecification/SSP-001"
              - const: "https://admin-shell.io/aas/API/3/0/AssetAdministrationShellRepositoryServiceSpecification/SSP-002"
              - const: "https://admin-shell.io/aas/API/3/0/SubmodelRepositoryServiceSpecification/SSP-001"
              - const: "https://admin-shell.io/aas/API/3/0/SubmodelRepositoryServiceSpecification/SSP-002"
              - const: "https://admin-shell.io/aas/API/3/0/SubmodelRepositoryServiceSpecification/SSP-003"
              - const: "https://admin-shell.io/aas/API/3/0/SubmodelRepositoryServiceSpecification/SSP-004"
              - const: "https://admin-shell.io/aas/API/3/0/ConceptDescriptionServiceSpecification/SSP-001"
              - {}
      type: object

```

**Comment by danielporta** [comment]
Ah @sebbader-sap, I probaby copied an old openapi version? 3.0.3 --> 3.1.0

**Comment by sebbader-sap** [comment]
Yep, you are right, "const" comes with OpenAPI V3.1.0, e.g.:

```
openapi: 3.1.0
info:
  title: MWE for anyOf validation
  description: a description
  contact:
    name: Industrial Digital Twin Association (IDTA)
    email: info@idtwin.org
  license: 
    name: CC BY 4.0
    url: https://creativecommons.org/licenses/by/4.0/
  version: V3.0.1_SSP-001
paths:
  /description:
    get:
      tags:
        - Description API
      summary: Returns the self-describing information of a network resource (ServiceDescription)
      operationId: GetDescription
      x-semanticIds:
        - https://admin-shell.io/aas/API/Descriptor/GetDescription/3/0
      responses:
        '200':
          description: Requested Description
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/ServiceDescription'
components: 
  schemas:   
    ServiceDescription: 
      description: The Description object enables servers to present their capabilities to the clients, in particular which profiles they implement. At least one defined profile is required. Additional, proprietary attributes might be included. Nevertheless, the server must not expect that a regular client understands them.
      example: >-
        {
          "profiles": [
            "https://admin-shell.io/aas/API/3/0/AssetAdministrationShellRegistryServiceSpecification/SSP-002",
            "https://admin-shell.io/aas/API/3/0/SubmodelRegistryServiceSpecification/SSP-002"
          ]
        }
      properties: 
        profiles:
          type: array
          minItems: 1
          items:
            type: string
            anyOf:
              - const: "https://admin-shell.io/aas/API/3/0/AssetAdministrationShellServiceSpecification/SSP-001"
              - const: "https://admin-shell.io/aas/API/3/0/AssetAdministrationShellServiceSpecification/SSP-002"
              - {}
      type: object

```

This one passes the validation.

**Comment by danielporta** [comment]
As discussed in the meeting, we propose to change the OpenAPI definition to the following: 

```
openapi: 3.1.0
info:
  title: DotAAS Part 2 | HTTP/REST | Asset Adminstration Shell Service Specification
  description: Description
  contact:
    name: Industrial Digital Twin Association (IDTA)
    email: info@idtwin.org
  license: 
    name: CC BY 4.0
    url: https://creativecommons.org/licenses/by/4.0/
  version: V3.0.1_SSP-001
paths:
  /description:
    get:
      tags:
        - Description API
      summary: Returns the self-describing information of a network resource (ServiceDescription)
      operationId: GetDescription
      x-semanticIds:
        - https://admin-shell.io/aas/API/Descriptor/GetDescription/3/0
      responses:
        '200':
          description: Requested Description
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/ServiceDescription'
components: 
  schemas:   
    ServiceDescription: 
      description: The Description object enables servers to present their capabilities to the clients, in particular which profiles they implement. At least one defined profile is required. Additional, proprietary attributes might be included. Nevertheless, the server must not expect that a regular client understands them.
      example: >-
        {
          "profiles": [
            "https://admin-shell.io/aas/API/3/1/AssetAdministrationShellRegistryServiceSpecification/SSP-002",
            "https://admin-shell.io/aas/API/3/0/SubmodelRegistryServiceSpecification/SSP-002"
          ]
        }
      properties: 
        profiles:
          type: array
          minItems: 1
          items:
            type: string
            minLength: 1
            maxLength: 2024
      type: object
```

**Comment by sebbader-sap** [comment]
@danielporta next attempt: Value of `profiles` is only `type: string`, however, I put all currently known profiles identifiers under `example`. Thereby, they are still in the file. My test with the openapi-validator tool did pass, also the rendering in Swaggerhub seems ok.

File: Part2-API-Schemas/openapi.yaml [hunk 1]
```diff
@@ -685,7 +685,7 @@ components:
           minItems: 1
           items:
             type: string
-            enum:
+            example:
```
Comment (BirgitBoss) [inline Part2-API-Schemas/openapi.yaml:688]: is the following list of profiles a correct example of an array of strings?

Comment (danielporta) [reply Part2-API-Schemas/openapi.yaml:688]: Well, the array example is under ServiceDescription, this example here lists the currently defined profile strings that can be used in an array. So, I would let the commit pass.

**Review by danielporta — APPROVED** [review]

**Review by BirgitBoss — APPROVED** [review]
Daniel checked my comments and answered by questions

**Pull request merged**